Bottom: 887271c46209e42f7413273b7d1dd9a145214c2d
Top:    e44b1792cc7517520c169f6a30c56f035c3e2e68
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-12 14:50:48 -0500

Refresh of add-buildbot-ptxdist-worker

---
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
index 9f917097..b84c5b8b 100644
--- a/roles/buildbot-worker-ptxdist/defaults/main.yml
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -40,7 +40,7 @@ container_name: '{{ docker_name_prefix }}{{ hostname }}'
 container_volumes:
 - '{{ data_volume }}:{{ data_path }}:z'
 - '{{ cache_volume }}:{{ cache_path }}:z'
-- '{{ config_hostdir }}:{{ secrets_path }}'
+- '{{ secrets_hostdir }}:{{ secrets_path }}'
 
 # mount path of the data volume
 data_path: '/buildbot'
@@ -68,7 +68,7 @@ image_dir: '{{ docker_projects_dir }}/{{ hostname }}'
 image_name: '{{ docker_registry_username }}/{{ hostname }}'
 
 # repository URI of the image source
-image_repo: https://github.com/novatechweb/docker-buildbot-worker-ptxdist.git
+image_repo: '{{ buildbot_worker_ptxdist_image_repo | default("https://github.com/novatechweb/docker-buildbot-worker-ptxdist.git") }}'
 
 # host path containing secrets
 secrets_hostdir: '{{ config_hostdir }}'
diff --git a/roles/buildbot-worker-ptxdist/tasks/bootstrap.sh b/roles/buildbot-worker-ptxdist/tasks/bootstrap.sh
deleted file mode 100755
index 8353ad9d..00000000
--- a/roles/buildbot-worker-ptxdist/tasks/bootstrap.sh
+++ /dev/null
@@ -1,17 +0,0 @@
-#!/bin/sh
-
-DIRS="/cache \
-      /cache/ccache \
-      /cache/downloads \
-      /cache/images \
-      /cache/ipkg-repository \
-      /cache/mirrors \
-      /cache/releases \
-      /cache/sstate"
-
-for d in $DIRS
-do
-  mkdir -p $d
-  chown -c -R buildbot:buildbot $d
-  chmod -v 777 $d
-done
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index 179cb373..f7a280c1 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -34,6 +34,16 @@
   docker_volume:
     name: '{{ data_volume }}'
 
+- name: Create secrets volume
+  file:
+    path: '{{ secrets_hostdir }}'
+    state: directory
+
+- name: Populate known_hosts secret
+  copy:
+    content: '{{ known_hosts }}'
+    dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
+
 - name: Checkout image repo
   git:
     repo: '{{ image_repo }}'
